lib/bundler/plugin/installer.rb

```diff
@@ -14,12 +14,10 @@ class Installer
       def install(names, options)
         version = options[:version] || [">= 0"]
         Bundler.settings.temporary(:lockfile_uses_separate_rubygems_sources => false, :disable_multisource => false) do
-          if options[:git]
-            install_git(names, version, options)
-          else
-            sources = options[:source] || Bundler.rubygems.sources
-            install_rubygems(names, version, sources)
-          end
+          sources = extract_sources_from_options!(options)
```

Comment:
Existing code prioritizes `--git` over `--source` - are we preserving this? I'm not sure if anyone was relying on respecting `--git` over `--source` where they were using both, but it is a behavioral change to add multiple sources to the `SourceList`.

Reply:
So! The rationale behind having a separate refactoring commit (e496b30b75) was in fact to make it clear what was happening. Such refactoring commit did not change the behavior.

This is what was happening before:

```ruby
def install_git(names, version, options)
  # [some stuff]

  source_list = SourceList.new
  source_list.add_git_source(options)

  # To support both sources
  if options[:source]
    source_list.add_rubygems_source("remotes" => options[:source])
  end

  # [some stuff]
end
```

the post-refactoring code is clearer (I hope :laughing:):

Reply:
(note: I've edited the comment due to a misplaced Ctrl+Enter :smile:)